src/java/org/apache/cassandra/index/sai/disk/v1/V1SearchableIndex.java

```diff
@@ -106,7 +107,9 @@ public V1SearchableIndex(SSTableContext sstableContext, IndexComponents.ForRead
             this.maxTerm = metadatas.stream().map(m -> m.maxTerm).max(TypeUtil.comparator(indexContext.getValidator(), version)).orElse(null);
 
             this.numRows = metadatas.stream().mapToLong(m -> m.numRows).sum();
-            this.approximateTermCount = metadatas.stream().mapToLong(m -> m.totalTermCount).sum();
+            this.approximateTermCount = metadatas.stream()
+                                                 .mapToLong(m -> m.totalTermCount == INVALID_TOTAL_TERM_COUNT ? 0 : m.totalTermCount)
+                                                 .sum();
```

Comment:
What is the behavior of a node having some total term counts of `INVALID_TOTAL_TERM_COUNT`? As an aside, I wonder if it is possible to compute a valid `totalTermCount` on start up for an old index.

Reply:
So how will we operate in a split scenario? Is it worth delaying usage of the total term count until we have it for all components? This is technically why we have the `IndexFeatureSet` object, and might be a good place to use it.

Reply:
The main risk here is in the very early migration. For example, the first time a memtable index gets a row, that row's term count will be the average doc length for the whole node.